container/navigation.go

```diff
@@ -0,0 +1,196 @@
+	s := nav.Title
+	if nav.level > 0 {
+		s = nav.titles[nav.level-1]
+	}
+	nav.PushWithTitle(obj, s)
+}
+
+// PushWithTitle puts the given CanvasObject on top, hides the object below, and uses the given title as label text.
+//
+// Since: 2.6
+func (nav *Navigation) PushWithTitle(obj fyne.CanvasObject, s string) {
+	objs := nav.stack.Objects[:nav.level]
+	if len(objs) > 0 {
+		objs[len(objs)-1].Hide()
+	}
+	nav.stack.Objects = append(objs, obj)
+	if len(nav.stack.Objects) > 0 {
+		if nav.Back != nil {
+			nav.Back.Enable()
+		}
+	}
+	nav.titles = append(nav.titles[:nav.level], s)
+	nav.level++
+	if nav.Label != nil {
+		nav.Label.SetText(s)
+	}
+	if nav.Next != nil {
+		nav.Next.Disable()
+	}
+	obj.Show()
+}
+
+// Pop return the top level CanvasObject, adjusts the title accordingly, and disabled the back button
+// when no more objects are left to go back to.
+//
+// Since: 2.6
+func (nav *Navigation) Pop() fyne.CanvasObject {
+	if nav.level == 0 || nav.level == 1 && nav.Root != nil {
+		return nil
+	}
+
+	title := nav.Title
+	objs := nav.stack.Objects
+	objs[nav.level-1].Hide()
+	if nav.level > 1 {
+		objs[nav.level-2].Show()
+		title = nav.titles[nav.level-2]
+	}
+	nav.Label.SetText(title)
+
+	if nav.level == 1 || nav.level == 2 && nav.Root != nil {
+		if nav.Back != nil {
+			nav.Back.Disable()
+		}
+	}
+
+	if nav.Next != nil {
+		nav.Next.Enable()
+	}
+	nav.level--
+
+	return objs[nav.level]
+}
+
+// Forward shows the next object in the again.
+//
+// Since: 2.6
+func (nav *Navigation) Forward() fyne.CanvasObject {
+	nav.stack.Objects[nav.level-1].Hide()
+	nav.stack.Objects[nav.level].Show()
+
+	s := nav.Title
+	if len(nav.titles) > 0 {
+		s = nav.titles[nav.level]
+	}
+	nav.level++
+	nav.Label.SetText(s)
+
+	if nav.level == len(nav.stack.Objects) {
+		if nav.Next != nil {
+			nav.Next.Disable()
+		}
+	}
+	if nav.level > 0 {
+		if nav.Back != nil {
+			nav.Back.Enable()
+		}
+	}
+
+	return nav.stack.Objects[nav.level-1]
+}
+
+// SetTitle changes the navigation title and the title for the current object.
+//
+// Since: 2.6
+func (nav *Navigation) SetTitle(s string) {
+	nav.titles[nav.level] = s
+	nav.Label.SetText(s)
+}
+
+func (nav *Navigation) setup() {
+	objs := []fyne.CanvasObject{}
+	titles := []string{}
+	if nav.Root != nil {
+		objs = append(objs, nav.Root)
+		titles = append(titles, nav.Title)
+	}
+	nav.level = len(objs)
+	nav.stack.Layout = layout.NewStackLayout()
+	nav.stack.Objects = objs
+	nav.titles = titles
+	nav.ExtendBaseWidget(nav)
+}
+
+func (nav *Navigation) CreateRenderer() fyne.WidgetRenderer {
+	nav.setup()
+	box := []fyne.CanvasObject{}
+	if nav.Back != nil {
+		box = append(box, nav.Back)
+	}
+	if nav.Next != nil {
+		box = append(box, layout.NewSpacer(), nav.Next)
+	}
+	control := NewStack(NewHBox(box...))
+	if nav.Label != nil {
+		control.Objects = append(control.Objects, nav.Label)
+	}
+	return widget.NewSimpleRenderer(NewBorder(control, nil, nil, nil, &nav.stack))
```

Comment:
With the comments above, I think we basically can't use the simple renderer. There needs to be an actual custom renderer with the internal widgets and properties reachable so we can have the methods on that renderer change the internal widgets when values on our widget is changed. You can use https://github.com/Jacalz/rymdport/blob/4707bc88f31670e2de0293f914f0894e30b78755/internal/ui/components/stack.go#L79 for inspiration.